src/wp-includes/user.php

```diff
@@ -585,8 +585,15 @@ function count_user_posts( $userid, $post_type = 'post', $public_only = false )
 	global $wpdb;
 
 	$where = get_posts_by_author_sql( $post_type, true, $userid, $public_only );
```

Comment:
Improve cache hits by sorting post types.

```suggestion
	$post_type = (array) $post_type;
	sort( $post_type );

	$where = get_posts_by_author_sql( $post_type, true, $userid, $public_only );
```